frontend/src/metabase/query_builder/actions.js

```diff
@@ -1370,8 +1375,54 @@ export const showChartSettings = createAction(SHOW_CHART_SETTINGS);
 export const onUpdateVisualizationSettings = updateCardVisualizationSettings;
 export const onReplaceAllVisualizationSettings = replaceAllCardVisualizationSettings;
 
-export async function createModerationReview(reviewParams) {
-  await ModerationReviewApi.create(reviewParams);
+export const CREATE_MODERATION_REVIEW = "metabase/qb/CREATE_MODERATION_REVIEW";
+export const createModerationReview = createThunkAction(
+  CREATE_MODERATION_REVIEW,
+  ({ moderationReview, moderationRequest }) => async (dispatch, getState) => {
+    const currentUser = getUser(getState());
+    const moderatorId = currentUser.id;
+    const { type, cardId, description } = moderationReview;
+    if (isRequestDismissal(type)) {
+      if (!moderationRequest) {
+        throw new Error("Missing moderation request -- unable to dismiss.");
+      }
+
+      await ModerationRequestApi.update({
+        id: moderationRequest.id,
+        status: "dismissed", // todo
+        closed_by_id: moderatorId,
+      });
+    } else {
+      const reviewPromise = ModerationReviewApi.create({
+        status: type,
+        moderated_item_id: cardId,
+        moderated_item_type: "card",
+        text: description,
+      });
+
+      let requestPromise;
+      if (moderationRequest) {
+        requestPromise = ModerationRequestApi.update({
+          id: moderationRequest.id,
+          status: "resolved", // todo
+          closed_by_id: moderatorId,
+        });
```

Comment:
Does the `closed_by_id` parameter mean that I can put any user ID in the request and make it look like somebody else has updated a moderation request? I'm pretty sure nobody is going to do that deliberately, but still wanted to double-check

Reply:
right now, yes, but I expect there will be additional validation on the BE with respect to that. nothing right now, but I think once we're more confident about the exact implementation it will get added.